doc/cookbook/source/examples/classifier/LDA.rst

```diff
@@ -0,0 +1,54 @@
+=============================
+Linear Discriminant Analysis
+=============================
+
+LDA learns a linear classifier and requires examples to be CDenseFeatures. The learned linear classification rule is optimal under the assumption that both classes a gaussian distributed with equal co-variance. To find a linear separation :math:`{\bf w}` in training, the in-between class variance is maximized and the within class variance is minimized.
+Three methods to compute the LDA:
+
+Two class Fisher Discriminant Analysis (FLD_LDA):
+
+.. math::
+
+	J({\bf w})=\frac{{\bf w^T} S_B {\bf w}}{{\bf w^T} S_W {\bf w}}
+
+Singular Valued decomposition method (SVD_LDA):  The above derivation of Fisher's LDA requires the invertibility of the within class matrix. However, this condition gets void when there are fewer data-points than dimensions. A solution is to require that :math:`{\bf W}` lies only in the subspace spanned by the data. A basis of the data :math:`{\bf X}` is found using the thin-SVD technique which returns an orthonormal non-square basis matrix :math:`{\bf Q}`. We then require the solution :math:`{\bf w}` to be expressed in this basis.
+
+.. math:: 
+
+	{\bf W} := {\bf Q} {\bf{W^\prime}}
+
+AUTO_LDA: This mode automagically chooses one of the above modes for the users based on whether N > D (chooses FLD_LDA) or N < D(chooses SVD_LDA)
+
+See Chapter 16 in :cite:`barber2012bayesian` for a detailed introduction.
+
+-------
+Example
+-------
+
+We create CDenseFeatures (here 64 bit floats aka RealFeatures) and :sgclass:`CBinaryLabels` from files with training and test data.
+
+.. sgexample:: LDA.sg:create_features
+
+We create an instance of the :sgclass:`CLDA` classifier and set features and labels. The default method is AUTO.
```

Comment:
By default, Shogun automatically chooses the decomposition method based on  :math:`{N<D}`